spot_driver/include/spot_driver/robot_state/spot_robot_state_publisher.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <spot_driver/interfaces/robot_state_client_interface.hpp>
+
+#include <spot_driver/interfaces/logger_interface_base.hpp>
+#include <spot_driver/interfaces/parameter_interface_base.hpp>
+#include <spot_driver/interfaces/tf_interface_base.hpp>
+#include <spot_driver/interfaces/timer_interface_base.hpp>
+
+#include <rclcpp/node.hpp>
+#include <tl_expected/expected.hpp>
+
+namespace spot_ros2 {
+
+/**
+ * @brief Retrieves robot state from Spot's robot state service client, converts the messages from Protobuf to ROS
+ * messages, and publishes them to the appropriate topics
+ */
+class SpotRobotStatePublisher {
+ public:
+  /**
+   * @brief A handle that enables dependency injection of ROS and rclcpp::node operations
+   */
+  class MiddlewareHandle {
+   public:
+    virtual ~MiddlewareHandle() = default;
+
+    virtual void createPublishers() = 0;
+    virtual void publishRobotState(const RobotState& robot_state) = 0;
+
+    virtual std::shared_ptr<rclcpp::Node> node() = 0;
```

Comment:
I find suspicious that we need to return an rclpp::Node, for the same reason of the previous comment.

Reply:
I refactored to remove this.